### DISCUSSION TITLE
Proxy reference changes unexpectedly without updates

### DISCUSSION THREAD

**Discussion by s524797336** [pmndrs/valtio#985] [Bug report]
### Bug Description

Since version 2.1.0, `state.obj` reference keeps changing unexpectedly, while it worked as expected in version 2.0.0. Is this a bug or a feature?

```typescript
import React from 'react'
import { proxy, useSnapshot } from 'valtio'

const state = proxy({
  count: 0,
  obj: {},
})

const TestComponent: React.FC = () => {
  useEffect(() => {
    const timer = setInterval(() => {
      state.count++
    }, 1000)
    return () => clearInterval(timer)
  }, [])

  const { obj } = useSnapshot(state)

  useEffect(() => {
    console.log('Why obj keep changing?', obj)
  }, [obj])

  return null
}
```

In this example, `state.obj`'s reference changes continuously in version 2.1.0, even though obj itself is not being updated. This issue was not present in version 2.0.0.

### Reproduction Link

null

**Comment by dai-shi** [comment]
Thanks for reporting. It's either #967 or #970.
Can you identify which causes it?
Try
```bash
pnpm add https://pkg.pr.new/valtio@967
```
or
```bash
pnpm add https://pkg.pr.new/valtio@970
```

**Reply by s524797336** [reply]
Cannot install using the given command; it returns a 404 error.

**Reply by s524797336** [reply]
I switched to using `pnpm add https://pkg.pr.new/valtio@8b7f7bf` for #967 and `pnpm add https://pkg.pr.new/valtio@c3af924` for #970.

#967 works fine, but #970 does not.

**Reply by dai-shi** [reply]
Thanks for looking into it. I'll work on it.

**Reply by dai-shi** [reply]
#986 reverts #970.
